### PR TITLE
[libtheora] add unofficial cmake config, remove FindOGG.cmake

### PR DESCRIPTION
**Pull request by mcmtroffaes**
- What does your PR fix? This adds cmake config file for unofficial::theora targets, to use by other ports and consumers of vcpkg. It also removes the FindOGG.cmake since libogg in vcpkg already provides a config file. Tested on x64-linux, x64-windows, and x64-windows-static.

- Which triplets are supported/not supported? Have you updated the CI baseline? No changes.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes.

### CODE REVIEW

File: ports/libtheora/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ file(GLOB HEADERS
 )
 
 include_directories("include")
-include_directories(${OGG_INCLUDE_DIR})
+include_directories(${Ogg_INCLUDE_DIR})
```
Comment (ras0219) [inline ports/libtheora/CMakeLists.txt:15]:
```suggestion
```

This should be handled automatically by the `Ogg::ogg` target.

**Review by ras0219 — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: ports/libtheora/CMakeLists.txt (hunk 1 above)
Comment (mcmtroffaes) [reply ports/libtheora/CMakeLists.txt:15]: Thanks, and yes, I realised that. I tried deleting it but for some reason it does not find the headers for some files that are compiled. I'm going to see if I can get to the root of this.

Comment (mcmtroffaes) [reply ports/libtheora/CMakeLists.txt]: Adding a target_link_libraries against Ogg for the object libraries fixed it. I've just pushed another commit to this effect.

Comment (PhoebeHui) [reply ports/libtheora/CMakeLists.txt]: The targets may need have target_include_directories(${target} PUBLIC $<INSTALL_INTERFACE:include>)

Comment (mcmtroffaes) [reply ports/libtheora/CMakeLists.txt]: Ah, good point. Might as well fix that too.

**Comment by mcmtroffaes** [comment]
Thanks @ras0219 @PhoebeHui for your comments. I think I've now done everything and done another series of tests to ensure it's all compiling as it should. Let me know if you still spot some issues.

I'll try to get this cmake file be included and maintained upstream. Not that theora still gets a lot of development, but it can't hurt trying, and it feels like the right thing to do.

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your updates!

**Comment by mcmtroffaes** [comment]
You're welcome. I've submitted the cmake work to upstream here:

https://github.com/xiph/theora/pull/14

Whilst I was at it, I've also submitted the patch by @Mixaill (which fixes the debug build) to upstream:

https://github.com/xiph/theora/pull/13

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**